### DISCUSSION TITLE
First-time user migrating data from YouTube: Subscriptions didn't load, history is mixed together, and playlist import requires a different format?

### DISCUSSION THREAD

**Discussion by CyberAkumaZero** [FreeTubeApp/FreeTube#4174] [Q&A]
I followed the instructions on the Docs page on how to download and import your history and subscriptions mentioned here:

https://docs.freetubeapp.io/usage/importing-history/

I downloaded the history, subscription, and playlist data for several of my YouTube accounts, creating a separate profile for each one on Freetube. I switched to each profile before importing subscriptions and history for it. However, when I go to subscriptions for any of my profiles they are empty even though it said they were successfully imported, I can only see ones I later manually added through the Freetube app.

It also appears to have just simply mashed the history of all of my profiles together regardless of which profile I select. When I first opened the history tab it was loading everything up for hours, eventually crashing and forcing a restart of the app. When I restarted the same extensive history appears for every profile.

I also noticed that there is a import playlists option as well, but it asks for them in a format YouTube does not support. Google Takeout only outputs them in .csv format while Freetube wants a .db format file.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @CyberAkumaZero, ill try to answer your questions.

> I downloaded the history, subscription, and playlist data for several of my YouTube accounts, creating a separate profile for each one on Freetube. I switched to each profile before importing subscriptions and history for it. However, when I go to subscriptions for any of my profiles they are empty even though it said they were successfully imported, I can only see ones I later manually added through the Freetube app.

That is indeed and unfortunately the correct behavior for FT. All of your subscription files will be imported into the All Channels Profile it does not matter if you switched profile beforehand. 

Related issue https://github.com/FreeTubeApp/FreeTube/issues/1996

Ill include a video for u to show u how u can organize your profiles properly


https://github.com/FreeTubeApp/FreeTube/assets/73130443/629258f8-4f78-479b-9986-31c4863aa231


> It also appears to have just simply mashed the history of all of my profiles together regardless of which profile I select. When I first opened the history tab it was loading everything up for hours, eventually crashing and forcing a restart of the app. When I restarted the same extensive history appears for every profile.

The History page is a global page across all profiles so it doesn't matter in what profile u watch a video. So all the history files you imported will be accessible on the History tab.

Related issue https://github.com/FreeTubeApp/FreeTube/issues/1996

> I also noticed that there is a import playlists option as well, but it asks for them in a format YouTube does not support. Google Takeout only outputs them in .csv format while Freetube wants a .db format file.

That is correct FT only accepts a .db at the moment because we dont have a fully implemented playlist feature yet. The .db file is what u get when u export the playlists within FT so u can import it on another machine running FT.

Related issue https://github.com/FreeTubeApp/FreeTube/issues/1016

I hope that your questions are answered by this but if something is unclear let me know! :)

**Reply by Julienist** [reply]
Okay so, if FT only accepts a .db file as playlist-format (at the moment of importing a playlist).
Then, how do people such as me convert a .csv file to a .db file?
Because, basically importing from YouTube as in .csv files into FT isn't an option.

I have seen some pages explaining how to do this with SQL (this is I think one way to do it, or I'm overthinking it – also a possibility).
But if this is one way of doing it with SQL,
Is it expected that users will tinker with their .csv files in something like SQL, and convert the containing data into a .db file?

If there is a simpler way of converting a .csv file (as in playlist) to a .db file?

Yes, I try to keep it simple.... TRYING.

**Reply by absidue** [reply]
FreeTube doesn't support local playlists yet, it only supports a list of favourites.

So even if you did somehow manage to convert the csv to a compatible format, you wouldn't see them or be able to interact with your playlists in any way in FreeTube.

That button is only for exporting and importing the list of favourites between different FreeTube installs.

**Reply by Julienist** [reply]
Okay, a follow-up question for this would be:
Is there support for local playlists coming (any time soon – I do not mind if it takes a long time, just curious)?
Even if there's no support coming any time soon, this question would be asked anyway.
If it's not known for there to be support coming for it, well that's also an answer to my question.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
V0.20.0😇

**Reply by Julienist** [reply]
aha, nice to know. thanks for the quick reply
